### ISSUE TITLE
Generation of in-line JS unsafe

### ISSUE THREAD

**Issue by LaughingJohn** [dotnet/AspNetCore.Docs#23283]
Generating in-line Javascript seems unsafe to me and also require the use of script-src 'unsafe-inline' in the CSP.

This can be made more secure by moving this code

```
new QRCode(document.getElementById("qrCode"),
            {
                text: "@Html.Raw(Model.AuthenticatorUri)",
                width: 150,
                height: 150
            });
```
to a separate javascript file and amending as follows:

```
window.onload = function () {
    const uri = document.getElementById("qrCodeData").getAttribute('data-url');
    new QRCode(document.getElementById("qrCode"),
        {
            text: uri,
            width: 150,
            height: 150
        });
};
```
This allows the removal of script-src 'unsafe-inline' .

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 117c950a-0bcb-d936-31cf-5859b4280abb
* Version Independent ID: 270bedc7-a636-5bd0-9e2f-e42235deec8c
* Content: [Enable QR Code generation for TOTP authenticator apps in ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/security/authentication/identity-enable-qrcodes?view=aspnetcore-5.0)
* Content Source: [aspnetcore/security/authentication/identity-enable-qrcodes.md](https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/security/authentication/identity-enable-qrcodes.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-security**
* GitHub Login: @Rick-Anderson
* Microsoft Alias: **riande**

**Comment by Rick-Anderson** [comment]
@blowdart 30 second review

**Comment by blowdart** [comment]
This is fair :) Works for me

**Comment by Rick-Anderson** [comment]
@LaughingJohn  
Any chance you could PR this? At the top right of the article, select the Edit pen icon:

![image](https://user-images.githubusercontent.com/3605364/53368931-11972200-38ee-11e9-86ce-18aaff69c719.png)

Select the pen icon again. Save.

If not, I'll fix it.

**Comment by LaughingJohn** [comment]
Thanks @Rick-Anderson. I've made an attempt to PR it (I've never done one before). I wanted to move some of the paragraphs around but didn't as I was afraid of breaking the moniker directives, please see my comment on the PR.

**Comment by Rick-Anderson** [comment]
I don't see a PR in our repo. I could fix the monikers before you PR so that wouldn't be a probelm.

> Thanks @Rick-Anderson. I've made an attempt to PR it (I've never done one before). I wanted to move some of the paragraphs around but didn't as I was afraid of breaking the moniker directives, please see my comment on the PR.

**Comment by LaughingJohn** [comment]
Sorry @Rick-Anderson, I have never used Github in anger, so have no idea what I'm doing. I *think* I've managed to create a PR here: https://github.com/dotnet/AspNetCore.Docs/pull/23306.

**Comment by serpent5** [comment]
Closed in #23306.
